### ISSUE TITLE
what is the structure of the LEXICON input ?

### ISSUE THREAD

**Issue by wassimhennoune** [bohana/sentlex#4]
what is the structure of the LEXICON input ?

**Comment by bohana** [comment]
Hello,

Source input data format to `Lexicon` objects is configurable. A common one is to read from a file ie. the `ResourceLexicon` you find here:
https://github.com/bohana/sentlex/blob/master/sentlex/sentlex.py#L286

A `ResourceLexicon` can then use a customized loader function to load the lexicon into memory. An example implementation can be seen for the `SWNLexicon':
https://github.com/bohana/sentlex/blob/master/sentlex/sentlex.py#L496

And the corresponding loader utility function:
https://github.com/bohana/sentlex/blob/master/sentlex/sentlexutil.py#L53

...which loads a corresponding file from here:
https://github.com/bohana/sentlex/tree/master/sentlex/data

In short, you can load a lexicon from whatever format you need, by writing your custom loader function.

**Comment by wassimhennoune** [comment]
thank you so much
